pkg/api/api.go

```diff
@@ -21,9 +21,9 @@ type Plugin struct {
 
 // Definition Represents an API that you want to proxy
 type Definition struct {
-	Name        string            `bson:"name" json:"name" valid:"required"`
+	Name        string            `bson:"name" json:"name" valid:"required~name is required"`
```

Comment:
please add integration tests to check if these errors are really go to a response

Reply:
Added, however adding validation and integration tests to all other fields like `upstreams.targets.target` is out of scope of this PR, cos ATM api allows inserting empty upstreams, it will be fixed here https://github.com/hellofresh/janus/issues/251